src/server/zset_family.cc

```diff
@@ -1340,6 +1342,55 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(smvec.size());
 }
 
+void ZSetFamily::ZInterCard(CmdArgList args, ConnectionContext* cntx) {
+  unsigned num_keys;
+  if (!absl::SimpleAtoi(ArgS(args, 0), &num_keys)) {
+    return (*cntx)->SendError(OpStatus::SYNTAX_ERR);
+  }
+
+  uint64_t limit = 0;
+  if (args.size() == 1 + num_keys + 2 && ArgS(args, 1 + num_keys) == "LIMIT") {
+    if (!absl::SimpleAtoi(ArgS(args, 1 + num_keys + 1), &limit)) {
+      return (*cntx)->SendError("limit value is not a positive integer", kSyntaxErrType);
+    }
+  } else if (args.size() != 1 + num_keys) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+
+  vector<OpResult<ScoredMap>> maps(shard_set->size(), OpStatus::SKIPPED);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    maps[shard->shard_id()] = OpInter(shard, t, "", AggType::NOOP, {}, false);
+    return OpStatus::OK;
+  };
+
+  cntx->transaction->Schedule();
+  cntx->transaction->Execute(std::move(cb), false);
```

Comment:
i am surprised it's working. 
you use `Execute(, false)` that does not finalize the transaction.
@chakaz  can you please deep dive into this? How come your last checks do not catch this?
@daniel-shimon for this operation ScheduleSingleHop should work better because you just need a single hop to fetch all the intersections  and you do not need another one (unlike with ...STORE)

Reply:
It fails for me:

```
[ RUN      ] ZSetFamilyTest.ZInterCard
F20230510 15:52:06.120864 380643 test_utils.cc:173] Check failed: NumLocked() == 0U (2 vs. 0) 
*** Check failure stack trace: ***
    @     0x560a54c0021f  google::LogMessage::Fail()
    @     0x560a54c00165  google::LogMessage::SendToLog()
    @     0x560a54bff93a  google::LogMessage::Flush()
    @     0x560a54c037b2  google::LogMessageFatal::~LogMessageFatal()
    @     0x560a538e74c3  dfly::BaseFamilyTest::TearDown()
    @     0x560a547c5c41  testing::internal::HandleSehExceptionsInMethodIfSupported<>()
    @     0x560a547be711  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @     0x560a54792e62  testing::Test::Run()
    @     0x560a5479389e  testing::TestInfo::Run()
    @     0x560a547941a5  testing::TestSuite::Run()
    @     0x560a547a40b7  testing::internal::UnitTestImpl::RunAllTests()
    @     0x560a547c6bd6  testing::internal::HandleSehExceptionsInMethodIfSupported<>()
    @     0x560a547bf81b  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @     0x560a547a281f  testing::UnitTest::Run()
    @     0x560a538e212e  RUN_ALL_TESTS()
    @     0x560a538e1e87  main
    @     0x7efe30123d90  (unknown)
    @     0x7efe30123e40  __libc_start_main
    @     0x560a53854975  _start
*** SIGABRT received at time=1683723126 on cpu 0 ***
PC: @     0x7efe30190a7c  (unknown)  pthread_kill
    @     0x560a54c863ae         64  absl::lts_20230125::WriteFailureInfo()
    @     0x560a54c865b8         96  absl::lts_20230125::AbslFailureSignalHandler()
    @     0x7efe3013c520  (unknown)  (unknown)
fish: Job 1, './zset_family_test' terminated by signal SIGABRT (Abort)
fish returned exit code 134
```

@daniel-shimon Have you tried running the test? Did it not fail for you?

Reply:
Yeah I ran the test and got this failure, haven't gotten to figuring it out yet. 
I did see that this means a transaction was not finalized, but missed the `Execute` argument. 
Will fix in upcoming push :)

Reply:
Fixed